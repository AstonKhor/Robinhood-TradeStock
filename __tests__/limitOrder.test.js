/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import LimitOrder from '../client/components/limitOrder.jsx';

describe('<Limit Order /> Component', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<LimitOrder />);
    expect(tree).toMatchSnapshot();
  });

  it('initiates state as expected', () => {
    const wrapper = shallow(<LimitOrder />);
    expect(wrapper.state().limitPrice).toBe(0);
    expect(wrapper.state().shares).toBe(0);
    expect(wrapper.state().expires).toBe('Good for Day');
  });

  // Make test for state change on click of expires button;
});
