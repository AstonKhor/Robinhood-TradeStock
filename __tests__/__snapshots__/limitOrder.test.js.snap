// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Limit Order /> Component matches the snapshot 1`] = `
Array [
  <div
    className="sc-bxivhb cyWHkv"
  >
    <label
      className="sc-bwzfXH dbwzZe"
    >
      Limit Price
    </label>
    <input
      className="sc-bdVaJa sc-htpNat dFFWgj"
      name="limitPrice"
      onChange={[Function]}
      placeholder="$0.00"
      step=".01"
      type="number"
    />
  </div>,
  <div
    className="sc-bxivhb cyWHkv"
  >
    <label
      className="sc-bwzfXH dbwzZe"
    >
      Shares
    </label>
    <input
      className="sc-bdVaJa sc-htpNat dFFWgj"
      name="shares"
      onChange={[Function]}
      placeholder="0"
      type="number"
    />
  </div>,
  <div
    className="sc-bxivhb cyWHkv"
  >
    <span
      className="sc-bZQynM iCwhxW"
    >
      Expires
    </span>
    <div
      className="sc-htoDjs hyvtCa"
      onClick={[Function]}
    >
      Good for Day
    </div>
  </div>,
]
`;
